src/App.tsx

```diff
@@ -1,14 +1,105 @@
 /* eslint-disable max-len */
-import React from 'react';
+/* eslint-disable no-console */
+import React, {
+  useCallback, useEffect, useMemo, useState,
+} from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
+import debounce from 'lodash/debounce';
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { Select } from './types/Select';
+
+export const filterTodos = (s: Select, q: string, t: Todo[]): Todo[] => {
+  const lowerAppliedQuery = q.toLowerCase();
+
+  return t
+    .filter(td => {
```

Comment:
very bad naming :) if I look at this code for first time it was hard to understand what happened here